### ISSUE TITLE
"The dependency tzinfo-data will be unused" on Ubuntu

### ISSUE THREAD

**Issue by terracotta210** [tzinfo/tzinfo-data#12]
When I run "rails new appname" command  I get this message 

> The dependency tzinfo-data (>= 0) will be unused by any of the platforms Bundler is installing for. Bundler is installing for ruby but the dependency is only for x86-mingw32, x86-mswin32, x64-mingw32, java. To add those platforms to the bundle, run `bundle lock --add-platform mingw, mswin, x64_mingw, jruby`.

I've tried 
`gem install tzinfo-data `

but it didn't help.
I've also tried `bundle lock --add-platform mingw, mswin, x64_mingw, jruby` as the message suggested but it only works for particular app and  every time I create a new app I see the message again.

Ubuntu 14.04, 
Ruby 2.3.3, Rails 5.0.1

**Comment by philr** [comment]
The message is coming from Bundler (bundler/bundler#5003). It is not an error and can be ignored.

When you create a new Ruby on Rails application, a `Gemfile` is created that includes the following line:

```ruby
gem 'tzinfo-data', platforms: [:mingw, :mswin, :x64_mingw, :jruby]
```

The purpose of this line is to include the tzinfo-data gem in the bundle on Windows to act as a source of time zone data. This gem is unnecessary on Ubuntu (and Unix-based systems in general) because the system includes time zone data that can be read directly by tzinfo.

If you want to get rid of the message you can do one of the following:

1. Remove the `platforms` option from the `gem 'tzinfo-data'` line in the `Gemfile` (and run `bundle update`). This will cause tzinfo to use tzinfo-data as its data source on all platforms instead of using the system time zone data.
2. Remove the `gem 'tzinfo-data'` line from the `Gemfile`. This will cause tzinfo to always try and use the system time zone data. A `TZInfo::DataSourceNotFound` exception will be raised if you try and run your app on Windows.
3. Run `bundle lock --add-platform mingw, mswin, x64_mingw, jruby` to add mingw, mswin, x64_mingw, and jruby to the list of platforms Bundler will include in the bundle.

Each of these options would need to be carried out for each Rails application you create.

**Comment by terracotta210** [comment]
It's good to know that this is not an error, thanks for help.

**Comment by cronwel** [comment]
Thank you philr, I was wondering( and I am a new to programming) how did you learn that? I see lots off folks know great things (like you in this post), but how did you acquire that knowledge? I hope to learn how to learn when it comes to programming. Any advice would be greatly appreciated.

**Comment by philr** [comment]
@cronwel I'm the author of tzinfo and tzinfo-data, so that's why I know how they can be used. The `Gemfile` syntax and `bundler` command are documented on the [Bundler documentation site](https://bundler.io/docs.html).

**Comment by lifesucx** [comment]
I read philir's answer, and thank you. It is nice to know that it is not an error message, but I am no longer able to (I am sorry if this is worded badly) program in the terminal anymore. My username and the '$' and the name of the directory I am in do not show up. I can type in the line and hit enter, but it is more like I am writing a paper now. No program is executing. I have tried exiting and reopening the terminal and running the program again, but it is always the same output.

**Comment by sptty** [comment]
I have the pros too,  I dont know how to fix it . so sad ...

**Comment by mxhold** [comment]
Just adding that an additional way to silence this warning is by enabling the `disable_platform_warnings` option for Bundler (available in versions >=1.17.0) by running:

```bash
bundle config --local disable_platform_warnings true
```

This will silence any platform warnings in the context of the current app. If you leave off the `--local` option, the warnings will be silenced globally for the current machine.
